src/test/java/edu/hm/hafner/util/SetTest.java

```diff
@@ -0,0 +1,132 @@
+        assertThatThrownBy(() -> new HashSet<Integer>(-5, (float) 0.75))
+                .isExactlyInstanceOf(IllegalArgumentException.class);
+        assertThatThrownBy(() -> new HashSet<Integer>(5, (float) -0.75))
+                .isExactlyInstanceOf(IllegalArgumentException.class);
+    }
+
+    @Test
+    void testConstructorCollection() {
+        assertThatThrownBy(() -> new HashSet<Integer>(null))
+                .isExactlyInstanceOf(NullPointerException.class);
+    }
+
+    @Test
+    void shouldIterate() {
+        HashSet<Integer> h = createFilledHashSet(1, 2);
+        Iterator<Integer> i = h.iterator();
+
+        assertThat(i.next()).as("First Iteration").isEqualTo(1);
+        assertThat(i.next()).as("Second Iteration").isEqualTo(2);
+        assertThat(i.hasNext()).as("No more values available").isFalse();
+    }
+
+    @Test
+    void shouldReturnSize() {
+        HashSet<Integer> h = createFilledHashSet(1, 2, 3);
+
+        assertThat(h.size())
+                .as("Size of filled HashSet")
+                .isEqualTo(3);
+
+        h.clear();
+
+        assertThat(h.size())
+                .as("Size of empty HashSet")
+                .isEqualTo(0);
+    }
+
+    @Test
+    void shouldBeEmpty() {
+        HashSet<Integer> h = createEmptyHashSet();
+        assertThat(h.size()).isEqualTo(0);
+    }
+
+    @Test
+    void shouldContain() {
+        HashSet<Integer> h = createFilledHashSet(1, 2, 3);
+
+        assertThat(h.contains(1)).as("Contained value").isTrue();
+        assertThat(h.contains(10)).as("Value not contained").isFalse();
+    }
+
+    @Test
+    void shouldAdd() {
+        HashSet<Integer> h = createEmptyHashSet();
+
+        assertThat(h.add(1)).as("1 added").isTrue();
+        assertThat(h.add(1)).as("Another 1 added").isFalse();
+        assertThat(h.contains(1)).isTrue();
+
+        assertThat(h.add(-1)).as("-1 added").isTrue();
+        assertThat(h.contains(-1)).isTrue();
+
+        assertThat(h.add(null)).as("Null added").isTrue();
+        assertThat(h.contains(null)).isTrue();
+
+        assertThat(h.add(0)).as("0 added").isTrue();
+        assertThat(h.contains(0)).isTrue();
+    }
+
+    @Test
+    void shouldRemove() {
+        HashSet<Integer> h = createFilledHashSet(1, 2, 3);
+        assertThat(h.contains(1)).isTrue();
+        assertThat(h.remove(1)).as("Remove value included").isTrue();
+        assertThat(h.contains(1)).isFalse();
+
+        assertThat(h.contains(0)).isFalse();
+        assertThat(h.remove(0)).as("Remove value not included").isFalse();
+        assertThat(h.contains(0)).isFalse();
+    }
+
+    @Test
+    void shouldClear() {
+        HashSet<Integer> h = createFilledHashSet(1, 2, 3);
+        assertThat(h.isEmpty()).isFalse();
+        h.clear();
+        assertThat(h.isEmpty()).isTrue();
+    }
+
+    @Test
+    void shouldClone() {
+        HashSet<Integer> h = createFilledHashSet(1, 2, 3);
+        Object g = h.clone();
+        assertThat(h.clone()).isEqualTo(g);
```

Comment:
Hier vergleichst du zwei geklonte Sets. Du willst aber das eigentliche Set mit einem geklonten vergleichen. Sonst wäre der Test sogar erfolgreich, wenn `clone` immer das gleiche zurückgibt.
```suggestion
        assertThat(h).isEqualTo(g);
```